### PR TITLE
Add documentation for custom group roles

### PR DESCRIPTION
**Pull request by renzol2**
This PR adds documentation for custom group roles, a new feature being implemented in #8006.

### CODE REVIEW

File: docs/assessment.md
Comment (mfsilva22) [inline docs/assessment.md:294]: the example above already has only "Recorder" with the permission to Submit. either change the example, or make this question level permission different

File: docs/assessment.md
Comment (mfsilva22) [inline docs/assessment.md:314]: can this image be smaller?

File: docs/assessment.md
Comment (mfsilva22) [inline docs/assessment.md:318]: this image still has the "Save" button enabled. Is this what was decided at the end? I thought Geoffrey Herman suggested that both buttons should be disabled.

Comment (renzol2) [reply docs/assessment.md:318]: No, you're correct in that we decided in the end that both buttons should be disabled. I used an outdated screenshot; I'll update that now. Thanks for pointing it out!

File: docs/assessment.md
Comment (renzol2) [reply docs/assessment.md:314]: I'm not sure if you'd like the literal image size to be smaller, or if the screenshot should be less zoomed in. I did the former, but let me know if you meant the latter or something else entirely!

Comment (mfsilva22) [reply docs/assessment.md:314]: I would prefer less zoom (the size of the fonts in the screenshot should be similar to the ones in the text). As it is now, the font size of the image is substantially larger than everything else.

**Review by mfsilva22 — APPROVED** [review]

File: docs/assessment.md
Comment (nwalters512) [inline docs/assessment.md:294]: It might be nice to show a stripped-down example of an assessment config with a concrete demonstration of setting permissions at the various levels.

File: docs/assessment.md
Comment (nwalters512) [inline docs/assessment.md:298]: This is not a dropdown menu in the usual sense - maybe say something like this instead?

```suggestion
When students join a group, they are automatically assigned a role. Students can always view the roles of other students in the group, both before and during an assessment. Students can click "View role info" to see more information about the assessment's group roles.
```

You'll need to change the usage of "dropdown menu" a few lines down as well.

File: docs/grouproles_submit_question.png
Comment (nwalters512) [inline docs/grouproles_submit_question.png:1]: Looking at this picture, it feels a little strange to have the lock icon placed between the two buttons. I'd consider moving it to the right of "Save only"? (I recognize this change will have to be made in the other PR, not here.)

Comment (renzol2) [reply docs/grouproles_submit_question.png:1]: Agreed. In hindsight, it was partly because we only disabled Save & Grade that we kept the lock icon in the middle, but with both controls being disabled, moving the lock icon to the right makes more sense. It was also partly because I couldn't figure out how to move it to the right, but TIL about `order-last` classes in Bootstrap!

**Review by nwalters512 — APPROVED** [review]
The new changes look good! I'll merge this once the group roles code itself is merged and deployed.

File: docs/grouproles_assign_roles.png
Comment (jonatanschroeder) [inline docs/grouproles_assign_roles.png:1]: The format of this dialog has changed a bit. TODO: create a new screenshot. Same for the other images.

File: docs/assessment.md
Comment (jonatanschroeder) [inline docs/assessment.md:233]:
```suggestion
- Assigning group roles for other students
```

File: docs/assessment.md
Comment (jonatanschroeder) [inline docs/assessment.md:235]:
```suggestion
Although in most cases each student is expected to take one role, students are allowed to take on multiple roles in some narrow scenarios, such as when users leave and join groups after assessments have started.
```

File: docs/assessment.md
Comment (jonatanschroeder) [inline docs/assessment.md:246]:
```suggestion
      "canAssignRoles": true
```

Assuming #9290 is merged.

File: docs/assessment.md
Comment (jonatanschroeder) [inline docs/assessment.md:270]:
```suggestion
| `canAssignRoles` | boolean | false   | Allow students with this role to assign roles to other students. |
```

Will require a re-formatting of the table.

File: docs/assessment.md
Comment (jonatanschroeder) [inline docs/assessment.md:272]:
```suggestion
Students typically select their roles before starting an assessment, but they can change their roles mid-assessment if needed. As a safeguard against invalid role configurations, PrairieLearn prevents students from viewing questions if a group's role configuration does not meet the instructor's specification.
```

File: docs/assessment.md
Comment (jonatanschroeder) [inline docs/assessment.md:269]:
```suggestion
| `minimum`               | integer | 0       | The minimum required number of students holding this role in the assessment.              |
| `maximum`               | integer | -       | The maximum required number of students holding this role in the assessment.              |
```

File: docs/assessment.md
Comment (jonatanschroeder) [inline docs/assessment.md:322]:
```suggestion
When a role configuration is invalid, which may occur when a user leaves the group or a new user joins the group, students become unable to see any questions until the roles are reviewed.
```

File: docs/assessment.md
Comment (jonatanschroeder) [reply docs/assessment.md]: Done in ef84fc5dab11a3a9468e32aa7aebdf22ca2664c6.

File: docs/assessment.md
Comment (jonatanschroeder) [reply docs/assessment.md:332]: @mfsilva22 can you check https://prairielearn--8375.org.readthedocs.build/en/8375/assessment/#assigning-roles-to-students to see if you're ok with the Zoom level used in these new screenshots?

File: docs/assessment.md
Comment (nwalters512) [inline docs/assessment.md:292]: I think this is true, and it's probably helpful to include so people don't think they need to set both these properties everywhere.

```suggestion
Setting either attribute to `[]` (empty array) means that **no role** can view/submit that part. Setting either attribute to `null` or omitting it entirely means that **every role** can view/submit that part.
```

**Review by nwalters512 — APPROVED** [review]
This looks good to me! I'd encourage you to wait for a final sign-off from @mfsilva22.

File: docs/assessment.md
Comment (jonatanschroeder) [reply docs/assessment.md:292]: I fear that this may cause confusion if this is considered in the context of a question where a default is set for a zone/assessment. I'll rewrite this to make it clearer.

Comment (nwalters512) [reply docs/assessment.md:292]: I realized shortly after submitting this review that yeah, this behavior depends on what's configured at parent levels. Given that, I think this is fine as-is, though I'm happy to review a revised version if you think you can make it more clear.

Comment (jonatanschroeder) [reply docs/assessment.md]: Rephrased in 956f8c47549aa9d8b45ff351078fc0f49da3fdb3.

File: docs/assessment.md
Comment (nwalters512) [inline docs/assessment.md:292]: Nit: removing the word "question" since these attributes can be set (or not) at any level.

```suggestion
Setting either attribute to `[]` (empty array) means that **no role** can view/submit that part. If neither attribute is set, it means that **every role** can view/submit that part.
```

Comment (nwalters512) [reply docs/assessment.md:292]: Maybe this is worse, but my CS brain interpreted the original the wrong way (though it's obviously the wrong interpretation if you think about it for 2 seconds):

```suggestion
Setting either `canView` or `canSubmit` to `[]` (empty array) means that **no role** can view or submit that part, respectively. If a question does not set the value of either attribute, it means that **every role** can view/submit that part.
```

Comment (nwalters512) [reply docs/assessment.md:292]: (I'll leave it up to you to merge my two suggestions 🙂)

Comment (jonatanschroeder) [reply docs/assessment.md:292]:
```suggestion
Setting either `canView` or `canSubmit` to `[]` (empty array) means that **no role** can view or submit that part of the assessment, respectively. If either attribute is not set, it means that **every role** has the permission associated to the attribute, i.e., any student with any role can view or submit that part of the assessment.
```

**Review by mfsilva22 — APPROVED** [review]
This looks great!

**Pull request merged**
